_includes/parse-server/third-party-auth.md

```diff
@@ -405,3 +407,168 @@ For more information about custom auth please see the examples:
+
+### Time-based One-Time passwords (TOTP)
+
+Time-based one-time passwords are considered best practise for multi-factor authentication, as SMS OTPs can be intercepted via [SS7](https://www.theguardian.com/technology/2016/apr/19/ss7-hack-explained-mobile-phone-vulnerability-snooping-texts-calls) or [sim swap](https://us.norton.com/blog/mobile/sim-swap-fraud) attacks. TOTP authentication can be configured using:
+
+```js
+{
+  auth: {
+    mfa: {
+      enabled: true,
+      options: ['TOTP'],
+      algorithm: 'SHA1',
+      digits: 6,
+      period: 30,
+    },
+  },
+}
+```
+
+To enable MFA for a user, the [OTPAuth](https://github.com/hectorm/otpauth) package can be used.
+
+First, create an TOTP object:
+
+```js
+const secret = new OTPAuth.Secret();
+const totp = new OTPAuth.TOTP({
+  algorithm: "SHA1",
+  digits: 6,
+  period: 30,
+  secret,
+});
+```
+
+Next, ask the user to add the TOTP code to their authenticator app:
+
+```js
+const uri = totp.toString();
+```
+
+This URI can also be scanned as a QR code, using the [QRCode](https://www.npmjs.com/package/qrcode) package.
+
+```js
+QRCode.toCanvas(document.getElementById("canvas"), uri);
+```
+
+Now, to confirm the user has correctly added the TOTP to their authenticator app, ask them to provide a valid code.
+
+```js
+const token = ""; // user inputted code
+await user.save({
+  authData: {
+    mfa: {
+      secret: secret.base32, // secret is from the TOTP object above
+      token, // token is generated from the users authenticator app
+    },
+  },
+});
+```
+
+Now, MFA will be enabled for the user. You can access recovery keys by:
+
+```js
+const recovery = user.get("authDataResponse");
+```
+
+It's also recommended to clear the authData from the client side:
+
+```js
+await user.fetch();
+```
+
+Now, when this user logs in, the will need to provide a valid MFA code:
+
+```js
+const login = async () => {
+  try {
+    await Parse.User.logIn(username, password);
+  } catch (e) {
+    if (e.message === 'Missing additional authData mfa') {
+      // show code input dialog here
+    }
+  }
+}
+const loginWithTOTP = async () => {
+  try {
+    await Parse.User.logInWithAdditionalAuth(username, password, {
+      mfa: // mfa code here
+    });
+  } catch (e) {
+    // display error
+  }
+}
+```
+
+### SMS One-Time passwords
+
+It is recommended to use TOTP MFA over SMS OTPs as SMS OTPs can be intercepted via [SS7](https://www.theguardian.com/technology/2016/apr/19/ss7-hack-explained-mobile-phone-vulnerability-snooping-texts-calls) or [sim swap](https://us.norton.com/blog/mobile/sim-swap-fraud) attacks.
+
+```js
+{
+  auth: {
+    mfa: {
+      enabled: true,
+      options: ['SMS'],
+      sendSMS(otp, mobileNumber) {
+        // Use an SMS service to send the SMS OTP
+      },
+      digits: 6,
+      period: 30,
+    },
+  },
+}
+```
+
+To enable SMS MFA for a user, first set the users' mobile number.
+
+```js
+await user.save({ authData: { mfa: { mobile: "+11111111111" } } });
```

Comment:
It seems there is no official standard by the ITU for example phone numbers, like example.com by the IANA. I'd suggest to stick with the most common example phone number which is `+15555555555`. Twilio [docs](https://www.twilio.com/docs/voice/trusted-calling-with-shakenstir#use-a-calltoken-when-forwarding-a-call) also use this primarily, while in examples with multiple phone numbers they use the digits 2 and 8.

> A caller (+12222222222) dials your Twilio phone number (+15555555555) and you need to forward the call to a new phone number (+18888888888).